### ISSUE TITLE
remove array and string usage from gulp.watch

### ISSUE THREAD

**Issue by contra** [gulpjs/gulp#977]
```js
gulp.watch('glob', ['task1', 'task2'])
gulp.watch('glob', 'task2')
```

will have to be changed to

```js
gulp.watch('glob', gulp.parallel('task1', 'task2'))
gulp.watch('glob', gulp.task('task2'))
```

This makes it a lot clearer what is going on.

Lots of documentation is going to need to be updated.

Semi-related: I also think we should discourage use of a watch task and just have people do it like this

```js
gulp.task('a', fn);
gulp.task('b', fn);

gulp.watch('glob', gulp.parallel('a', 'b'));
```

and if they have a flag for when to watch it can be as simple as

```js
gulp.task('a', fn);
gulp.task('b', fn);

if (process.env.NODE_ENV !== 'production) {
  gulp.watch('glob', gulp.parallel('a', 'b'));
}
```

I feel like this would simplify a lot of things

**Comment by phated** [comment]
:+1: on the discouraging of task watches, especially because you get all the benefit of returned streams and stuff now (with gulp.series/parallel)

**Comment by dinoboff** [comment]
I don't think any doc example is using array to define the tasks gulp.watch should run. [The only case I can find is](https://github.com/gulpjs/gulp/blob/4.0/docs/API.md#tasks-2):

> When `tasks` is an array, the tasks will be run in parallel:
> ```
> gulp.watch('*.js', [one, two]);
> // is equivalent to
> gulp.watch('*.js', gulp.parallel(one, two));
> ```

Maybe it should mention that it's ease migration and is deprecated.

**Comment by dinoboff** [comment]
About discouraging use of a watch, I don't follow. The README gulpfile example watch task is:
```js
gulp.task(watch);
[...]
function watch() {
  gulp.watch(paths.scripts, scripts);
  gulp.watch(paths.images, images);
}
```
What are suggesting instead?

**Comment by phated** [comment]
@dinoboff we are going to remove the array syntax altogether since 4.0 allows breaking changes.

The README example would be switched to:

```js
if (process.env.NODE_ENV !== 'production) {
  gulp.watch(paths.scripts, scripts);
  gulp.watch(paths.images, images);
}
```

I actually prefer:

```js
if (process.argv.indexOf('watch') !== -1) {
  gulp.watch(paths.scripts, scripts);
  gulp.watch(paths.images, images);
}
```

@contra thoughts?

**Comment by contra** [comment]
@phated I think the NODE_ENV example is clearer IMO but doesn't matter too much

**Comment by dinoboff** [comment]
I see. It's a little more work for the doc then.

Which ever idiom, opt-in would be better:
```
if (process.argv.indexOf('--watch') !== -1) {
  gulp.watch(paths.scripts, scripts);
  gulp.watch(paths.images, images);
}
```
or
```
if (process.env.NODE_ENV === 'development') {
  gulp.watch(paths.scripts, scripts);
  gulp.watch(paths.images, images);
}
```

**Comment by kevva** [comment]
Example number two as there might be confusion about what the `--watch` flag is in the first example.

**Comment by dinoboff** [comment]
Are you sure about it (the no watch task part). IMO gulpfile.js are meant to define task, not to start anything, especially not a task that doesn't end.

How would it work with gulp cli flags like `gulg --tasks`?

**Comment by phated** [comment]
The docs were removed by @dinoboff and I required the task to be a function in https://github.com/gulpjs/gulp/commit/ba548241e5049dadfda9cd85d9ef0e72500bbce4
